src/dialect.rs

```diff
@@ -0,0 +1,55 @@
+use crate::allocator::{Allocator, NodePtr};
+use crate::cost::Cost;
+use crate::reduction::Response;
+use crate::run_program::{OperatorHandler, PreEval, RunProgramContext};
+
+pub struct Dialect {
+    quote_kw: Vec<u8>,
+    apply_kw: Vec<u8>,
+    op_handler: Box<dyn OperatorHandler>,
```

Comment:
I don't see why this needs to be a run-time interface instead of a compile-time trait, not why it needs to be heap allocated. In the chia dialect case this class only contains one bool of state (strict mode).

`Dialect` seems like a perfect candidate for being a trait rather than a concrete type.